### PR TITLE
Fix too many arguments and remove broken includes

### PR DESCRIPTION
**Pull request by Maroc-OS**

### CODE REVIEW

File: src/common/serialization.tcc [hunk 1]
```diff
@@ -76,7 +76,7 @@ T reserialize(const T &obj)
 template<typename T>
 std::ostream& operator<<(std::ostream& out, const std::vector<T> &v)
 {
-    static_assert(!std::is_same<T, bool>::value, "this does not work for std::vector<bool>");
+    static_assert(!(std::is_same<T, bool>::value), "this does not work for std::vector<bool>");
```
Comment (tromer) [inline src/common/serialization.tcc:79]: Why is this needed? How can is `!std::is_same<T, bool>::value` misparsed?

Comment (ebfull) [reply src/common/serialization.tcc:79]: I've seen this before. It depends on the compiler, but the angle brackets for the templates screw with the parser.

Comment (ebfull) [reply src/common/serialization.tcc:79]: (`static_assert` is a macro.)

**Comment by daira** [comment]
We are no longer maintaining zcash/libsnark; closing all PRs.
